pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchActivityInstanceRoute.java

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.ddp.route;
+
+import org.apache.http.HttpStatus;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.ActivityInstanceDao;
+import org.broadinstitute.ddp.json.PatchSectionPayload;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.RouteUtil;
+import org.broadinstitute.ddp.util.ValidatedJsonInputRoute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+
+public class PatchActivityInstanceRoute extends ValidatedJsonInputRoute<PatchSectionPayload> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PatchActivityInstanceRoute.class);
+
+    private org.broadinstitute.ddp.db.ActivityInstanceDao activityInstanceDao;
+
+    public PatchActivityInstanceRoute(org.broadinstitute.ddp.db.ActivityInstanceDao activityInstanceDao) {
+        this.activityInstanceDao = activityInstanceDao;
+    }
+
+    @Override
+    public Object handle(Request request, Response response, PatchSectionPayload payload) throws Exception {
+        String userGuid = request.params(RouteConstants.PathParam.USER_GUID);
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+        String instanceGuid = request.params(RouteConstants.PathParam.INSTANCE_GUID);
+
+        LOG.info("Request to update section index on instance {} for participant {} in study {}",
+                instanceGuid, userGuid, studyGuid);
+
+        TransactionWrapper.useTxn(handle -> {
+            RouteUtil.findAccessibleInstanceOrHalt(response, handle, userGuid, studyGuid, instanceGuid);
+            int sectionsSize = activityInstanceDao.getActivityInstanceSectionsSize(handle, userGuid, studyGuid, instanceGuid);
```

Comment:
Is there a way to find `sectionsSize` without doing another database query? I think we can grab this number from the activity definition, which should have already been cached in memory. I believe we have some code to do this in PR #282. Can you work with @MocanaAtBroad to see if we can integrate that here?

Reply:
@yufengwng would this give us the right value? 
https://github.com/broadinstitute/ddp-study-server/blob/9b7b64da097d880e394b8568b77dda3bb8958533/pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/definition/FormActivityDef.java#L40

Reply:
I don't think so, because there may be an introduction or closing section (we will want to count those as well right?). We can call `getAllSections().size()` or along those lines.